src/ui/components/PrefixBadge.tsx

```diff
@@ -20,6 +20,7 @@ export default function PrefixBadgeButton({ breakpoint }: { breakpoint: Breakpoi
 
   return (
     <PrefixBadgePicker
+      key={breakpoint.options.prefixBadge}
```

Comment:
Without this, external changes to the badge color (aka ones made from the context menu) won't be rendered and the two will seem out of sync.

Note that this change currently _breaks_ the click-to-open interaction for this picker because it had a pre-existing issue where clicking unexpectedly _cleared_ the selected badge (which now means a new `key` and resetting its is-open internal state).

Reply:
Makes sense, key is a nice escape hatch here! 😅 I didn't look too much into how we could use derived state so we're not relying on local badge state which I carried over from the previous implementation.

Reply:
IMO there are some valid use cases for derived state, but [many times it's not necessary](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html). I _think_ this is one of those times (not necessary) but I didn't want to assume too much before talking to you about why it was the way it was.